### DISCUSSION TITLE
Plotting separate datasets upon one pole figure.

### DISCUSSION THREAD

**Discussion by beroshcr** [mtex-toolbox/mtex#2026] [Ask Anything]
Hi!

Assuming that two separate datasets are from the same sample and are spatially aligned, is it possible to plot the crystal directions (Ex. [001]) for an arbitrary mineral from both datasets onto one, combined pole figure (for the [001]) direction? 

Would appreciate any insight or help regarding this. 

Christian

**Comment by kilir** [comment]
Hi,

orientation * symmetrise(Miller) gives you the direction in specimen coordinates as vector3d. You can plot as many as you want on top of each other.

Hope that helps.

Cheers,
Rüdiger

**Reply by beroshcr** [reply]
Hi Rüdiger,

I haven't yet been able to get this working. I'm not the most skilled when it comes to coding unfortunately. But, when I do get this working would I, in theory, be able to contour all the plotted data on the pole figure? Regardless if they're from two separate data sets? 

For context, I am working on an EBSD paper regarding plotting the crystal orientations of a geological sample. The reviewers would like to see a larger sample size of grains examined. However, due to technical limitations with the laboratory I am working with, they are unable to do sample-scale indexing, and thus I am restricted to indexing small, individual areas of the sample. I am ultimately attempting to find a way to group several small areas of EBSD analysis to produce a larger data set.

 Me figuring this out will determine if this manuscript is accepted. Any advice or additional assistance you could provide would be extremely appreciated. 

Thank you.

Christian

**Comment by lfgmorales** [comment]
Hi christian

Not sure I understood your question properly but assuming you have 2 ebsd maps, you can calculate the odfs of them separately, then concatenate them together by writing something like

odf_total=[odf_1, odf_2]

And then you can plot odf_total as the two maps concatenate in one set of pole figures you have defined

Let me know if that works

Cheers

Luiz



Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 9 Jan 2024, at 04:24, beroshcr ***@***.***> wrote:

﻿

Hi Rüdiger,

I haven't yet been able to get this working. I'm not the most skilled when it comes to coding unfortunately. But, when I do get this working would I, in theory, be able to contour all the plotted data on the pole figure? Regardless if they're from two separate data sets?

For context, I am working on an EBSD paper regarding plotting the crystal orientations of a geological sample. The reviewers would like to see a larger sample size of grains examined. However, due to technical limitations with the laboratory I am working with, they are unable to do sample-scale indexing, and thus I am restricted to indexing small, individual areas of the sample. I am ultimately attempting to find a way to group several small areas of EBSD analysis to produce a larger data set.

Me figuring this out will determine if this manuscript is accepted. Any advice or additional assistance you could provide would be extremely appreciated.

Thank you.

Christian

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2026#discussioncomment-8060199>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN4CAMUYF3EDKDXQUTLYNSZ63AVCNFSM6AAAAABBNODYR2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DANRQGE4TS>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2026#discussioncomment-8060199", "url": "https://github.com/mtex-toolbox/mtex/discussions/2026#discussioncomment-8060199", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by beroshcr** [reply]
Hey Luiz, thank you for that, it worked. 

However, I am required to plot grain orientations (reconstructed from the ebsd maps), rather than ebsd.orientations.  

Would it be possible concatenate the orientations of reconstructed grains generated from the two separate ebsd maps? 

Christian

**Comment by lfgmorales** [comment]
Hi

Reconstruct grains, calculate odf of the grain mean orientation and concatenate the odfs

Pay attention however if that makes sense, in certain applications you are required to have the full dataset (for example, physical properties dont care about “grains”)

Cheers

Luiz
Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 9 Jan 2024, at 23:20, beroshcr ***@***.***> wrote:

﻿

Hey Luiz, thank you for that, it worked.

However, I am required to plot grain orientations (reconstructed from the ebsd maps), rather than ebsd.orientations.

Would it be possible concatenate the orientations of reconstructed grains generated from the two separate ebsd maps?

Christian

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2026#discussioncomment-8072284>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5INZIHFQBLDXQ2N3YMUTYNW7ANAVCNFSM6AAAAABBNODYR2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DANZSGI4DI>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2026#discussioncomment-8072284", "url": "https://github.com/mtex-toolbox/mtex/discussions/2026#discussioncomment-8072284", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by beroshcr** [comment]
Hi Luiz,

I used the same method suggested for concatenating the odfs to concatenate the two different reconstructed grain data sets and it seemingly worked? It at least produced my desired outcome. Assuming that these ebsd datasets are spatially oriented in the same way is there anything else you might consider prudent to ensure data integrity? Is there any obvious errors I might be overlooking? 

Do you think there could be an issue here when calculating the odf for these combined data sets when computing the J-index? 

My work flow:
```matlab
ebsd1 = EBSD.load([mtexEBSDPath filesep 'M60206 1 Specimen 1 rom line 2 Site 1 Map Data 1.crc'],...
    'convertEuler2SpatialReferenceFrame', 'setting 2');

ebsd2 = EBSD.load([mtexEBSDPath filesep 'M60206 1 Specimen 1 rom line 2 Site 2 Map Data 2.crc'],...
    'convertEuler2SpatialReferenceFrame', 'setting 2');

% reconstruct grains with a threshold angle of 10 degrees
grains1 = calcGrains(ebsd1('indexed'),'theshold',10*degree)

grains1 = grains1(grains1.grainSize > 10);

% smooth the grains to avoid the staircase effect
grains1 = smooth(grains1,5);

plot(grains1)

grains1

% reconstruct grains with a theshold angle of 10 degrees
grains2 = calcGrains(ebsd2('indexed'),'theshold',10*degree)

grains2 = grains2(grains2.grainSize > 10);

% smooth the grains to avoid the staircase effect
grains2 = smooth(grains2,5);

plot(grains2)

grains2

AuOri1 = grains1('augite1200006').meanOrientation
CSau = AuOri.CS

AuOri3  = grains3('augite1200006').meanOrientation
CSau = AuOri.CS

grains_total=[grains1,grains2]

AuOritotal = grains_total('augite1200006').meanOrientation
CSau = AuOri.CS

hAu = Miller({1,0,0,'uvw'},{0,1,0,'uvw'},{0,0,1,'uvw'}, CSau);
plotPDF(AuOritotal, hAu, 'contourf', 'all','halfwidth',10*degree)
```
